src/cephadm/cephadm.py

```diff
@@ -3082,6 +3082,18 @@ def get_container_mounts(ctx, fsid, daemon_type, daemon_id,
         data_dir = get_data_dir(fsid, ctx.data_dir, daemon_type, daemon_id)
         mounts.update(cc.get_container_mounts(data_dir))
 
+    # Modifications podman makes to /etc/hosts causes issues with
+    # certain daemons (specifically referencing "host.containers.internal" entry
+    # being added to /etc/hosts in this case). To avoid that, but still
+    # allow users to use /etc/hosts for hostname resolution, we can
+    # mount the host's /etc/hosts file.
+    # https://tracker.ceph.com/issues/58532
+    # https://tracker.ceph.com/issues/57018
+    if isinstance(ctx.container_engine, Podman):
+        if os.path.exists('/etc/hosts'):
+            if '/etc/hosts' not in mounts:
+                mounts['/etc/hosts'] = '/etc/hosts'
```

Comment:
> ... worried about something the container does messing with the host's actual /etc/hosts file, but there isn't any actual examples of that I can think of so that's probably fine.

Can this mount be made read-only just in case?  E.g. something like:
```suggestion
                mounts['/etc/hosts'] = '/etc/hosts:ro'
```

Reply:
you're right, we definitely should be doing that. Added that in.